dataset/src/main/scala/frameless/TypedDatasetForwarded.scala

```diff
@@ -130,6 +256,38 @@ trait TypedDatasetForwarded[T] { self: TypedDataset[T] =>
   def union(other: TypedDataset[T]): TypedDataset[T] =
     TypedDataset.create(dataset.union(other.dataset))
 
+  /**
+    * Randomly splits this [[TypedDataset]] with the provided weights.
+    * Weights for splits, will be normalized if they don't sum to 1.
+    *
+    * apache/spark
+    */
+  // $COVERAGE-OFF$ We can not test this method because it is non-deterministic.
+  def randomSplit(weights: Array[Double]): Array[TypedDataset[T]] =
```

Comment:
Should this also be wrapped into a `F: SparkDelay[F]`?

Reply:
@OlivierBlanvillain I'm not sure, it creates new ```Datasets``` but it doesn't seem to materialize it in any way.

When I run it in ```spark-shell``` it doesn't show any stages. What's the correct way to see whether given method triggers execution?

Reply:
Not sure, I guess one way to know is to load a large dataset and see how long the method takes to return. But if you tried it on spark-shell it should be good enough, anyway that's not hard to change after the fact :smile: